File: lista-de-exercicios/exercicio-5/Tabuada.ts
export class Tabuada {
    private numero: number;

    public constructor (numero: number) {
        this.numero = numero;
    }

    // Calcular que retorna um array com os resultados
    public calcular() {
        const resultados: Array<number> = [];

        if (!this.numero)
            throw Error ('Número inválido');
        for (let i = 1; i <= 10; i ++) {
            let resultado: number = this.numero * i;
            resultados.push(resultado);
        }
        return resultados;
    }

    // Calcular que retorna um array com as fórmulas e resultados
    public calcularSt() {
        const resultados: Array<string> = [];

        if (!this.numero)
            throw Error ('Número inválido');
        for (let i = 1; i <= 10; i ++) {
            let resultado: string = `${this.numero}x${i}=` + this.numero * i;
            resultados.push(resultado);
        }
        return resultados;
    }
}

File: lista-de-exercicios-java/src/main/java/org/example/exercicio05/Tabuada.java
package org.example.exercicio05;

import java.util.Arrays;

public class Tabuada {

    public String calcular(int numero) {
        int[] resultados = new int[10];

        for (int i = 0; i < 10; i ++) {
            int resultado = numero * (i + 1);
            resultados[i] = resultado;
        }
        return Arrays.toString(resultados);
    }

    public String calcularSt(int numero) {
        StringBuilder sb = new StringBuilder();

        for (int i = 1; i <= 10; i ++)
            sb.append(numero).append("x").append(i).append("=").append(numero * i).append("\n");
        return sb.toString();
    }
}
